integreat_cms/static/src/js/machine-translation-overlay.ts

```diff
@@ -0,0 +1,216 @@
+        console.log(page.title);
+        const list = document.createElement("li");
+        list.classList.add("list-disc", "ml-4");
+        list.textContent = page.title;
+
+        if (previewListOfPages.children.length <= limitOfPreview) {
+            previewListOfPages.appendChild(list);
+        } else {
+            listOfOptionalPages.appendChild(list);
+        }
+
+        numberOfTranslatablePages += 1;
+        numberOfWords += page.words;
+    });
+
+    document.getElementById("machine-translation-overlay-pages-number").textContent =
+        numberOfTranslatablePages.toString();
+    document.getElementById("machine-translation-overlay-budget-usage-result").textContent = numberOfWords.toString();
+    calculateAndSetRemainingBudget();
+};
+
+const buildNotTranslatableList = (notTranslatablePages: PageAttributes, expandableWarningListElement: HTMLElement) => {
+    const pages = Object.values(notTranslatablePages);
+
+    let numberOfNotTranslatablePages = 0;
+
+    pages.forEach((page) => {
+        const list = document.createElement("li");
+        list.classList.add("list-disc", "ml-4");
+        list.textContent = page.title;
+        expandableWarningListElement.appendChild(list);
+        numberOfNotTranslatablePages += 1;
+    });
+
+    document.getElementById("not-translateable").textContent = numberOfNotTranslatablePages.toString();
+};
+
+const prepareOverlay = (filteredPages: Pages) => {
+    const previewListOfPages = document.getElementById("machine-translation-overlay-pages");
+    previewListOfPages.textContent = "";
+    const listOfOptionalPages = document.getElementById("machine-translation-overlay-pages-optional");
+    listOfOptionalPages.textContent = "";
+    buildTranslatableList(filteredPages.translatable, previewListOfPages, listOfOptionalPages);
+
+    const warningListElement = document.getElementById("machine-translation-overlay-warning");
+    const expandableWarningListElement = document.getElementById("machine-translation-overlay-warning-optional");
+    expandableWarningListElement.textContent = "";
+    buildNotTranslatableList(filteredPages.not_translatable, expandableWarningListElement);
+
+    const expansionTrigger = document.getElementById("machine-translation-overlay-expansion-trigger");
+    if (previewListOfPages.children.length === 0) {
+        document.getElementById("machine-translation-overlay-pages-no-pages-warning").classList.remove("hidden");
+    } else if (previewListOfPages.children.length > limitOfPreview) {
+        expansionTrigger.classList.add("block");
+        expansionTrigger.classList.remove("hidden");
+    } else {
+        expansionTrigger.classList.add("hidden");
+        expansionTrigger.classList.remove("block");
+    }
+    toggleMachineTranslationForPagesButton();
+
+    if (expandableWarningListElement.children.length !== 0) {
+        warningListElement.classList.add("block");
+        warningListElement.classList.remove("hidden");
+    }
```

Comment:
```suggestion
    const expansionTrigger = document.getElementById("machine-translation-overlay-expansion-trigger");
    const noPageWarning = document.getElementById("machine-translation-overlay-pages-no-pages-warning");

    if (previewListOfPages.children.length === 0) {
        noPageWarning.classList.remove("hidden");
    } else {
        noPageWarning.classList.add("hidden");
    }

    if (listOfOptionalPages.children.length === 0) {
        expansionTrigger.classList.add("hidden");
        expansionTrigger.classList.remove("block");
    } else {
        expansionTrigger.classList.remove("hidden");
        expansionTrigger.classList.add("block");
    }

    toggleMachineTranslationForPagesButton();

    if (expandableWarningListElement.children.length !== 0) {
        warningListElement.classList.add("block");
        warningListElement.classList.remove("hidden");
    } else {
        warningListElement.classList.remove("block");
        warningListElement.classList.add("hidden");
    }
```

The warning test "None of the selected pages fullfills the required HIX...." was not toggled correctlly.